packages/main/src/AvatarGroup.js

```diff
@@ -0,0 +1,463 @@
+		this._hiddenItems = 0;
+		this._onResizeHandler = this._onResize.bind(this);
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get template() {
+		return AvatarGroupTemplate;
+	}
+
+	static get styles() {
+		return AvatarGroupCss;
+	}
+
+	static get dependencies() {
+		return [
+			Button,
+		];
+	}
+
+	/**
+	 * Returns an array containing the ui5-avatar instances that are currently not displayed due to lack of space.
+	 * @readonly
+	 * @type { Array }
+	 * @defaultValue []
+	 * @public
+	 */
+	get hiddenItems() {
+		return this.items.slice(this._hiddenStartIndex);
+	}
+
+	/**
+	 * Returns an array containing the <code>AvatarBackgroundColor</code> values that correspond to the avatars in the <code>items</code> array.
+	 * @readonly
+	 * @type { Array }
+	 * @defaultValue []
+	 * @public
+	 */
+	get colorScheme() {
+		return this.items.map(avatar => avatar._effectiveBackgroundColor);
+	}
+
+	get _hiddenStartIndex() {
+		return this._itemsCount - this._hiddenItems;
+	}
+
+	get _overflowBtnHidden() {
+		return this._hiddenItems === 0;
+	}
+
+	get _isGroup() {
+		return this.type === AvatarGroupType.Group;
+	}
+
+	get _itemsCount() {
+		return this.items.length;
+	}
+
+	get _groupTabIndex() {
+		return this._isGroup ? "0" : "-1";
+	}
+
+	get _overflowButtonTabIndex() {
+		return this._isGroup ? "-1" : false;
+	}
+
+	get _overflowButton() {
+		return this.shadowRoot.querySelector(AVATAR_GROUP_OVERFLOW_BTN_SELECTOR);
+	}
+
+	/**
+	 * Return the effective overflow button width
+	 * Differences are that when in "Group" type the button is offset and overlaps the avatars
+	 *
+	 * 1) In case of "Group", (LTR/RTL aware) button width is qual to second item offset left/right
+	 * 2) In case of "Individual" group type width is directly taken from button element
+	 * @private
+	 */
+	get _overflowButtonEffectiveWidth() {
+		// if in "Group" mode overflow button size is equal to the offset from second item
+		if (this._isGroup) {
+			let item = this.items[1];
+
+			// in some cases when second avatar is overflowed the offset of the button is the right one
+			if (!item || item.hidden) {
+				item = this._overflowButton;
+			}
+
+			return this.effectiveDir === "rtl" ? this._getWidthToItem(item) : item.offsetLeft;
+		}
+
+		return this._overflowButton.offsetWidth;
+	}
+
+	onAfterRendering() {
+		this._overflowItems();
+	}
+
+	onBeforeRendering() {
+		if (this._isGroup) {
+			this._itemNavigation.getItemsCallback = () => [];
+		} else {
+			this._itemNavigation.getItemsCallback = () => this.items.slice(0, this._hiddenStartIndex);
+		}
+
+		this._prepareAvatars();
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._onResizeHandler);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._onResizeHandler);
+	}
+
+	_onResize() {
+		this._overflowItems();
+	}
+
+	_onkeydown(event) {
+		if (isSpace(event)) {
+			// prevent scrolling
+			event.preventDefault();
+		}
```

Comment:
same as previous comment, I think Enter should be handled onkeydown, Space - onkeyup